lib/convection/model/mixin/policy.rb

```diff
@@ -92,10 +94,12 @@ def render
               'Effect' => effect,
               'Action' => action,
               'Resource' => resource
-            }.tap do |statemant|
-              statemant['Sid'] = sid unless sid.nil?
-              statemant['Condition'] = condition unless condition.nil?
-              statemant['Principal'] = principal unless principal.nil?
+            }.tap do |statement|
+              statement['Sid'] = sid unless sid.nil?
+              statement['Condition'] = condition unless condition.nil?
+              statement['Principal'] = principal unless principal.nil?
+
+              statement.delete('Resource') if resource.empty? # Avoid failure in CF if empty Resources array is passed
```

Comment:
Instead of deleting this key after the fact, move it's definition into the `tap` block, like:

```ruby
...
  'Effect' => effect,
  'Action' => action
}.tap do |statement|
  ...
  statement['Resource'] = resource unless resource.empty?
end
```